chrome/content/zotero/integration/quickFormat.js

```diff
@@ -1524,9 +1524,12 @@ var Zotero_QuickFormat = new function () {
 	/**
 	 * Called when progress changes
 	 */
-	function _onProgress(percent) {
+	async function _onProgress(percent) {
 		var meter = document.querySelector(".citation-dialog.progress-meter");
-		if(percent === null) {
+		meter.style.display = "";
+		await Zotero.Promise.delay(10);
+		_resizeWindow(meter.parentElement.parentElement.getBoundingClientRect().height);
```

Comment:
For me, the deck (`meter.parentElement.parentElement`) retains the height of the `editor`. If there are many lines of bubbles, `meter.parentElement.parentElement.getBoundingClientRect().height` is the same as the current height, so this line does not change the window height for me.

<img width="796" alt="Screenshot 2025-01-29 at 9 43 16 AM" src="https://github.com/user-attachments/assets/f55d75e7-ded5-4e0d-8e0d-fc18bdc186d1" />

Do we want to resize the window to only have enough space for the progress bar? We could run `editor.replaceChildren()` in `accept` to remove all bubbles after the citation object is updated to let `editor` (and the parent deck) go back to its original single-line height?

Reply:
I'm generally not happy with the need to rely on the XUL deck here, since it seems to replace the items in some async way. I guess it's slower when there's more elements. But it's easy to switch to a more reliable approach.